src/FsCodec.SystemTextJson/Pickler.fs

```diff
@@ -2,33 +2,13 @@
 
 open System.Text.Json
 
-[<AutoOpen>]
-module private Prelude =
-    let memoize (f: 'T -> 'S): 'T -> 'S =
-        let cache = new System.Collections.Concurrent.ConcurrentDictionary<'T, 'S>()
-        fun t -> cache.GetOrAdd(t, f)
-
 [<AbstractClass>]
 type JsonPickler<'T>() =
     inherit Serialization.JsonConverter<'T>()
 
-    static let isMatchingType =
```

Comment:
@eiriktsarpalis Apologies for yet more atting but I don't think I can answer it without you...

1. what desirable behaviors am I likely to be removing/breaking with this?
2. can you point me to anything that implements similar behavior, perhaps with some tests/docs/use cases?

My thinking is that I should have removed it long ago, as it does not fail the tests, and I can't think of a use case...

i.e. am I missing anything regarding some polymorphic behavior that's going over my head?

Reply:
Assuming you only use it to deserialize values (which is a naturally covariant operation) and never to serialize then it should work expected.

Reply:
Hm can you expand on that please? i.e. can you paint me a scenario as to how you do that, composing something with multiple pickers or something? i.e. what sort of a global hook would it afford one?

Reply:
Assuming you have `Foo <: IFoo` and you define a converter `JsonConverter<Foo>` whose `CanConvert` method returns `true` when given `typeof<IFoo>`, then the serializer will use that as the converter for `IFoo`. The downside however is that it will fail with a cast exception if you ever attempt to serialize values of type `IFoo` (since they may not be castable to `Foo`).

Reply:
Thanks! The main thing is that there's a pretty clear extension point for people to override the default behavior
(and it matches the general expectations one would have from STJ in terms of not being greedy by default)

Thankfully I've the luxury of not thinking too deeply about polymorphism wrt serialization in my day to day work ;)

Also now (V3 RC) is the right time to take out the behavior, if it's ever going to be done...